mat/mul_test.go

```diff
@@ -218,6 +218,24 @@ func (m *basicMatrix) T() Matrix {
 	return Transpose{m}
 }
 
+type basicVector VecDense
```

Comment:
Maybe these should all go in a separate file in future? (not just `basicVector`)

Reply:
Yes, that's exactly what I was thinking too.